app/models/user.rb

```diff
@@ -3,4 +3,21 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  validates :username, uniqueness: true
+  before_create :set_nickname
+
+  def email_required?
+    false
+  end
+
+  def will_save_change_to_email?
+    false
+  end
```

Comment:
Tried removing this and still was able to create a user and update a nickname, since no longer made the email required I think we should just remove this.